internal/dag/gatewayapi_processor.go

```diff
@@ -41,9 +43,59 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 		p.source = nil
 	}()
 
+	// Gateway must be defined for resources to be processed.
+	if p.source.gateway == nil {
+		p.Error("Gateway is not defined!")
+		return
+	}
+
 	for _, route := range p.source.httproutes {
-		p.computeHTTPRoute(route)
+
+		var validRoutes []*gatewayapi_v1alpha1.HTTPRoute
+
+		// Filter the HTTPRoutes that match the gateway which Contour is configured to watch.
+		// RouteBindingSelector defines a schema for associating routes with the Gateway.
+		// If Namespaces and Selector are defined, only routes matching both selectors are associated with the Gateway.
+
+		// ## RouteBindingSelector ##
+		//
+		// Selector specifies a set of route labels used for selecting routes to associate
+		// with the Gateway. If this Selector is defined, only routes matching the Selector
+		// are associated with the Gateway. An empty Selector matches all routes.
+		for _, listener := range p.source.gateway.Spec.Listeners {
+
+			matches, err := matchRoutes(listener.Routes.Selector, route.Labels)
+			if err != nil {
+				p.Errorf("error validating routes against Listener.Routes.Selector: %s", err)
+			}
+			if matches {
+				// Empty Selector matches all routes.
+				validRoutes = append(validRoutes, route)
```

Comment:
can we break out of the listeners loop, once we've found that the route matches at least one?

Reply:
Yeah for now I think that's fine.